rewriter/Prop.cc

```diff
@@ -38,6 +38,67 @@ bool isTStruct(ast::Expression *expr) {
     return struct_ != nullptr && struct_->cnst == core::Names::Constants::Struct() && isT(struct_->scope.get());
 }
 
+bool isTInexactStruct(ast::Expression *expr) {
+    auto *struct_ = ast::cast_tree<ast::UnresolvedConstantLit>(expr);
+    return struct_ != nullptr && struct_->cnst == core::Names::Constants::InexactStruct() && isT(struct_->scope.get());
+}
+
+bool isChalkODMDocument(ast::Expression *expr) {
+    auto *document = ast::cast_tree<ast::UnresolvedConstantLit>(expr);
+    if (document == nullptr || document->cnst != core::Names::Constants::Document()) {
+        return false;
+    }
+    auto *odm = ast::cast_tree<ast::UnresolvedConstantLit>(document->scope.get());
+    if (odm == nullptr || odm->cnst != core::Names::Constants::ODM()) {
+        return false;
+    }
+    auto *chalk = ast::cast_tree<ast::UnresolvedConstantLit>(odm->scope.get());
+    if (chalk == nullptr || chalk->cnst != core::Names::Constants::Chalk()) {
+        return false;
+    }
+    return chalk != nullptr && chalk->cnst == core::Names::Constants::Chalk() && isRootScope(chalk->scope.get());
```

Comment:
```suggestion
    return chalk->cnst == core::Names::Constants::Chalk() && isRootScope(chalk->scope.get());
```

Reply:
Good catch; actually the whole `if` is redundant.